rcon/maps.py

```diff
@@ -0,0 +1,849 @@
+        "hill400": "HIL",
+        "foy": "FOY",
+        "kursk": "KUR",
+        "stalingrad": "STA",
+        "remagen": "REM",
+        "kharkov": "KHA",
+        "driel": "DRL",
+        "elalamein": "ELA",
+    }
+)
+
+# Sourced with some minor modifications from https://github.com/timraay/Gamewatch/blob/master/
+
+
+class Gamemode(str, Enum):
+    WARFARE = "warfare"
+    OFFENSIVE = "offensive"
+    CONTROL = "control"
+    PHASED = "phased"
+    MAJORITY = "majority"
+
+    @classmethod
+    def large(cls):
+        return (
+            cls.WARFARE,
+            cls.OFFENSIVE,
+        )
+
+    @classmethod
+    def small(cls):
+        return (
+            cls.CONTROL,
+            cls.PHASED,
+            cls.MAJORITY,
+        )
+
+    def is_large(self):
+        return self in Gamemode.large()
+
+    def is_small(self):
+        return self in Gamemode.small()
+
+
+class Team(str, Enum):
+    ALLIES = "Allies"
+    AXIS = "Axis"
+
+
+class Environment(str, Enum):
+    DAY = "Day"
+    DUSK = "Dusk"
+    DAWN = "Dawn"
+    NIGHT = "Night"
+
+
+class Faction(str, Enum):
+    US = "us"
+    GER = "ger"
+    RUS = "rus"
+    GB = "gb"
+    CW = "gb"
+
+
+class Map(pydantic.BaseModel):
+    id: str
+    name: str
+    tag: str
+    prettyname: str
+    shortname: str
+    allies: "Faction"
+    axis: "Faction"
+
+    def __str__(self) -> str:
+        return self.id
+
+    def __repr__(self) -> str:
+        return str(self)
+
+    def __hash__(self) -> int:
+        return hash(self.id)
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, (Map, str)):
+            return str(self) == str(other)
+        return NotImplemented
+
+
+class Layer(pydantic.BaseModel):
+    id: str
+    map: Map
+    gamemode: Gamemode
+    attackers: Union[Team, None] = None
+    environment: Environment = Environment.DAY
+
+    def __str__(self) -> str:
+        return self.id
+
+    def __repr__(self) -> str:
+        return str(self)
+
+    def __hash__(self) -> int:
+        return hash(self.id)
+
+    def __eq__(self, other) -> bool:
+        if isinstance(other, (Layer, str)):
+            return str(self) == str(other)
+        return NotImplemented
+
+    def pretty(self):
+        out = self.map.prettyname
+        if self.gamemode == Gamemode.OFFENSIVE:
+            out += " Off."
+            if self.attackers:
+                out += f" {self.attackers.value.upper()}"
+        elif self.gamemode.is_small():
+            # TODO: Remove once more Skirmish modes release
+            out += " Skirmish"
+        else:
+            out += f" {self.gamemode.value.capitalize()}"
+        if self.environment != Environment.DAY:
+            out += f" ({self.environment.value})"
+        return out
+
+    @property
+    def opposite_side(self) -> Literal[Team.AXIS, Team.ALLIES] | None:
+        if self.gamemode == Gamemode.OFFENSIVE:
+            return Team.AXIS if self.attackers == Faction.GER else Team.ALLIES
```

Comment:
I honestly don't remember if I changed `attackers` from `Faction` to `Team` or whether you did, but either way, now that it is `Team`, this will always return `Team.ALLIES`

Reply:
I don't think it does actually, works as expected (but I do prefer your updated `attacking_faction` property so I'm going to swap it all over to that.

```python
@pytest.mark.parametrize(
    "team, expected",
    [
        (Team.ALLIES, Team.AXIS),
        (Team.AXIS, Team.ALLIES),
    ],
)
def test_get_opposite_side(team, expected):
    assert get_opposite_side(team) == expected
```

Reply:
Oh lol I am dumb and bad at reading, I was testing the utility function for it I made and not the property. I'm going to just update the property to use the properly working function which we use in the votemap settings anyway.